### PR TITLE
fix: entry list location fixes

### PR DESCRIPTION
**Pull request by bgutsol**
# Purpose of PR

- Merge master to canary
- Entry list location fixes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,18 @@
 
 - add entry list location ([bf72726](https://github.com/contentful/ui-extensions-sdk/commit/bf727265e3872b75c1f02eb940b5879fe31c7781))
 
+## [4.3.7](https://github.com/contentful/ui-extensions-sdk/compare/v4.3.6...v4.3.7) (2022-04-13)
```
Comment (andipaetzold) [inline CHANGELOG.md:7]: Can you reorder those by date?

File: lib/entry-list.ts [hunk 2]
```diff
@@ -68,30 +68,42 @@ const validateResult = (result: OnEntryListUpdatedHandlerReturn) => {
...
+  values: (value) => {
+    if (typeof value !== 'object') {
+      throw new Error(`Entry List location app data is invalid: 'values' should be an object.`)
+    }
+
+    const values = Object.values(value as Record<string, unknown>)
+    if (values.length === 0) {
+      // we allow passing empty values object
+      return
+    }
```
Comment (andipaetzold) [inline lib/entry-list.ts:84]: This check is not needed as calling `.every()` on an empty array is always true

File: lib/entry-list.ts [hunk 3]
```diff
@@ -68,30 +68,42 @@ const validateResult = (result: OnEntryListUpdatedHandlerReturn) => {
...
 
-const schema: Record<string, (value: unknown) => boolean> = {
-  values: (value) =>
-    typeof value === 'object' &&
-    Object.keys(value as Record<string, unknown>).length > 0 &&
-    Object.values(value as Record<string, unknown>).every((item) => typeof item === 'string'),
+const schema: Record<string, (value: unknown) => void> = {
+  values: (value) => {
+    if (typeof value !== 'object') {
+      throw new Error(`Entry List location app data is invalid: 'values' should be an object.`)
```
Comment (andipaetzold) [inline lib/entry-list.ts:77]:
```suggestion
      throw new Error(`Entry List location app data is invalid: 'values' must be an object.`)
```

File: lib/entry-list.ts [hunk 4]
```diff
@@ -68,30 +68,42 @@ const validateResult = (result: OnEntryListUpdatedHandlerReturn) => {
...
+    const values = Object.values(value as Record<string, unknown>)
+    if (values.length === 0) {
+      // we allow passing empty values object
+      return
+    }
+
+    const areValuesValid = values.every((item) => typeof item === 'string')
+    if (!areValuesValid) {
+      throw new Error(
+        `Entry List location app data is invalid: 'values' object should have a values of type 'string'.`
```
Comment (andipaetzold) [inline lib/entry-list.ts:89]:
```suggestion
        `Entry List location app data is invalid: 'values' object should only have values of type 'string'.`
```

**Review by andipaetzold — APPROVED** [review]
Some nits, but otherwise good.

File: lib/entry-list.ts (hunk 2 above)
Comment (bgutsol) [reply lib/entry-list.ts:84]: Nice catch ☝️

**Pull request merged**